### PR TITLE
fix: allow custom value for ipv6_cidr_block

### PR DESCRIPTION
**Pull request by cbeti-ionos**
## What does this fix or implement?

Allows user supplied values other than `AUTO` for the `ipv6_cidr_block` field of the nic resource.
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: ionoscloud/resource_nic.go [hunk 1]
```diff
@@ -294,6 +294,11 @@ func getNicData(d *schema.ResourceData, path string) ionoscloud.Nic {
 		}
 	}
 
+	if v, ok := d.GetOk(path + "ipv6_cidr_block"); ok {
+		ipv6_block := v.(string)
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_nic.go:298]: ipv6Block

File: docs/resources/nic.md [hunk 2]
```diff
@@ -27,6 +27,13 @@ resource "ionoscloud_ipblock" "example" {
     name                = "IP Block Example"
 }
 
+resource "ionoscloud_lan" "example"{
```
Comment (cristiGuranIonos) [inline docs/resources/nic.md:30]: best make this a separate example that uses ipv6

Comment (cbeti-ionos) [reply docs/resources/nic.md:29]: ipv6 example added separately

File: ionoscloud/resource_nic_test.go [hunk 3]
```diff
@@ -236,6 +238,18 @@ resource ` + constant.NicResource + ` "database_nic" {
   firewall_type = "BIDIRECTIONAL"
   ips = [ ionoscloud_ipblock.test_server.ips[0], ionoscloud_ipblock.test_server.ips[1] ]
   name = "updated"
+  ipv6_cidr_block = cidrsubnet(ionoscloud_lan.test_lan_2.ipv6_cidr_block,16,12)
+  ipv6_ips = [
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_nic_test.go:242]: indentation

File: ionoscloud/resource_nic_test.go [hunk 4]
```diff
@@ -236,6 +238,18 @@ resource ` + constant.NicResource + ` "database_nic" {
...
   name = "updated"
+  ipv6_cidr_block = cidrsubnet(ionoscloud_lan.test_lan_2.ipv6_cidr_block,16,12)
+  ipv6_ips = [ 
+				cidrhost(cidrsubnet(ionoscloud_lan.test_lan_2.ipv6_cidr_block,16,12),1),
+				cidrhost(cidrsubnet(ionoscloud_lan.test_lan_2.ipv6_cidr_block,16,12),2),
+				cidrhost(cidrsubnet(ionoscloud_lan.test_lan_2.ipv6_cidr_block,16,12),3)
+			]
+}
+
+data ` + constant.NicResource + ` test_nic_data {
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_nic_test.go:249]: indentation

File: ionoscloud/resource_lan_test.go [hunk 5]
```diff
@@ -145,7 +146,13 @@ resource ` + constant.LanResource + ` ` + constant.LanTestResource + ` {
   public = false
   name = "` + constant.UpdatedResources + `"
   pcc = ` + constant.PCCResource + `.` + constant.PCCTestResource + `.id
-}`
+  ipv6_cidr_block = cidrsubnet(` + constant.DatacenterResource + `.` + constant.DatacenterTestResource + `.ipv6_cidr_block` + `,8,2)
+}
+data ` + constant.LanResource + ` ` + constant.LanDataSourceById + ` {
+	datacenter_id = ` + constant.DatacenterResource + `.` + constant.DatacenterTestResource + `.id
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_lan_test.go:152]: indentation

File: ionoscloud/data_source_nic_test.go [hunk 6]
```diff
@@ -72,6 +77,12 @@ func Test_dataSourceNicRead(t *testing.T) {
 	if *nic.Properties.Dhcp != data.Get("dhcp").(bool) {
 		t.Fatalf("expected '%t', got '%s'", *nic.Properties.Dhcp, data.Get("dhcp"))
 	}
+	if *nic.Properties.Dhcpv6 != data.Get("dhcpv6").(bool) {
+		t.Fatalf("expected '%t', got '%s'", *nic.Properties.Dhcp, data.Get("dhcp"))
+	}
+	if *nic.Properties.Ipv6CidrBlock != data.Get("ipv6_cidr_block").(string) {
+		t.Fatalf("expected '%t', got '%s'", *nic.Properties.Dhcp, data.Get("dhcp"))
```
Comment (cristiGuranIonos) [inline ionoscloud/data_source_nic_test.go:84]: wrong err message, it's not dhcp that's being checked

File: ionoscloud/data_source_nic_test.go [hunk 7]
```diff
@@ -72,6 +77,12 @@ func Test_dataSourceNicRead(t *testing.T) {
 	if *nic.Properties.Dhcp != data.Get("dhcp").(bool) {
 		t.Fatalf("expected '%t', got '%s'", *nic.Properties.Dhcp, data.Get("dhcp"))
 	}
+	if *nic.Properties.Dhcpv6 != data.Get("dhcpv6").(bool) {
+		t.Fatalf("expected '%t', got '%s'", *nic.Properties.Dhcp, data.Get("dhcp"))
```
Comment (cristiGuranIonos) [inline ionoscloud/data_source_nic_test.go:81]: wrong err message, it's not dhcp that's being checked

File: ionoscloud/resource_lan_test.go [hunk 8]
```diff
@@ -70,6 +70,7 @@ func TestAccLanBasic(t *testing.T) {
 					resource.TestCheckResourceAttr(constant.LanResource+"."+constant.LanTestResource, "name", constant.UpdatedResources),
 					resource.TestCheckResourceAttr(constant.LanResource+"."+constant.LanTestResource, "public", "false"),
 					resource.TestCheckResourceAttrPair(constant.LanResource+"."+constant.LanTestResource, "pcc", constant.PCCResource+"."+constant.PCCTestResource, "id"),
+					resource.TestCheckResourceAttrPair(constant.DataSource+"."+constant.LanResource+"."+constant.LanDataSourceById, "ipv6_cidr_block", constant.LanResource+"."+constant.LanTestResource, "ipv6_cidr_block"),
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_lan_test.go:73]: indentation look weird here

File: ionoscloud/data_source_nic_test.go (hunk 7 above)
Comment (cbeti-ionos) [reply ionoscloud/data_source_nic_test.go]: corrected typo

File: ionoscloud/data_source_nic_test.go (hunk 6 above)
Comment (cbeti-ionos) [reply ionoscloud/data_source_nic_test.go]: corrected typo

**Pull request merged**